Ocaml native code generation for hppa has a bug
It can produce too big assembler files (>64k) that are invalid
See bug : http://bugs.gentoo.org/show_bug.cgi?id=178256
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=342704
http://caml.inria.fr/mantis/view.php?id=4147
http://caml.inria.fr/mantis/view.php?id=3937

This bug seems to have always existed but was never encountered
In 3.09 releases, ocamlp4 has grown bigger and bigger and triggers it
We just tell ocamlopt to optimize for size rather than for speed
and the produced assembler is thus valid...

Index: ocaml-3.09.3-patched/camlp4/config/Makefile.tpl
===================================================================
--- ocaml-3.09.3-patched.orig/camlp4/config/Makefile.tpl
+++ ocaml-3.09.3-patched/camlp4/config/Makefile.tpl
@@ -37,7 +37,7 @@ TEST_DIRECTORY=test `basename "$<"` = "$
 .ml.cmx:
 	@$(TEST_DIRECTORY)
 	@$(CAMLP4_COMM) $< -o $*.ppo
-	$(OCAMLOPT) $(OCAMLCFLAGS) -c -impl $*.ppo
+	$(OCAMLOPT) $(OCAMLCFLAGS) -compact -inline 0 -c -impl $*.ppo
 	rm -f $*.ppo
 
 .ml.p.cmx:
